### DISCUSSION TITLE
Help with error `'alag(.)'/'rate(.)'/'dur(.)' cannot depend on the state values`

### DISCUSSION THREAD

**Discussion by billdenney** [nlmixr2/rxode2#499] [Q&A]
I have a model with IV infusion data.  There is an issue with my dataset that I'm trying to track down.  Can you please help me understand what this error means?

The error is:

```
nlmixr(model_1cmt_linear, data = d_mod, est = "rxSolve")
# Error: 'alag(.)'/'rate(.)'/'dur(.)' cannot depend on the state values
```

I can't share the data, but I also don't see what the issue is with the data.

The model is a one-compartment IV model, and I'm pretty sure that the issue is not there:

```
model_1cmt_linear <- function() {
  description <- "One compartment PK model with linear clearance using differential equations"
  ini({
    lcl <- log(10) ; label("Linear clearance at high concentrations (L/hr)")
    lvc  <- log(5) ; label("Central volume of distribution (L)")
    cpaddSd <- 10 ; label("Additive residual error (pg/mL)")
    cppropSd <- 0.5 ; label("Proportional residual error (fraction)")
  })
  model({
    cl <- exp(lcl)
    vc  <- exp(lvc)

    kel <- cl / vc

    d/dt(central) <- -kel*central

    cp <- central / vc * 1000
    cp ~ add(cpaddSd) + prop(cppropSd)
  })
}
```

**Comment by mattfidler** [comment]
The error means that any of the parameters are directly or indirectly depending on the state values like  in the example above `central`.  The model you shared does not have those items so it could be that your data code for `rate` at `-1` or `-2`?  If so are you missing this definition in your model?

I believe the way this checks is to set all state values to `NA` and check to see if the output of the derived lag/dur/rate funtions give an NA (without the due diligence step of checking the output code).  If you have the rate defined in the data as a modeled rate and it is missing in the model perhaps it could also give the same error.
